content/nats-ca-rotation.md

```diff
@@ -0,0 +1,228 @@
+  path: /variables/-
+  value:
+    name: nats_ca_2
+    type: certificate
+    options:
+      is_ca: true
+      common_name: default.nats-ca.bosh-internal
+
+- type: replace
+  path: /variables/-
+  value:
+    name: nats_server_tls_2
+    type: certificate
+    options:
+      ca: nats_ca_2
+      common_name: default.nats.bosh-internal
+      alternative_names: [((internal_ip))]
+      extended_key_usage:
+      - server_auth
+
+- type: replace
+  path: /variables/-
+  value:
+    name: nats_clients_director_tls_2
+    type: certificate
+    options:
+      ca: nats_ca_2
+      common_name: default.director.bosh-internal
+      extended_key_usage:
+      - client_auth
+
+- type: replace
+  path: /variables/-
+  value:
+    name: nats_clients_health_monitor_tls_2
+    type: certificate
+    options:
+      ca: nats_ca_2
+      common_name: default.hm.bosh-internal
+      extended_key_usage:
+      - client_auth
+
+```
+
+### Step 2: Recreate all VMs, for each deployment. {: #step-2}
+
+Deployed VMs need to be recreated in order to receive new client certificates that are signed by the new CA. Also, they will receive a new list of CAs (old and new CAs certs concatenated) to trust when communicating with the NATS server. This recreation of the VMs is crucial for the NATS CA rotation.
+
+For example:
+
+```shell
+$ bosh -d deployment-name recreate
+```
+
+
+### Step 3: Update the director, health monitor, and NATS server jobs, to remove references for the old NATS CA and certificates signed by it. {: #step-3}
+
+```shell
+$ bosh create-env ~/workspace/bosh-deployment/bosh.yml \
+ --state ./state.json \
+ -o ~/workspace/bosh-deployment/[IAAS]/cpi.yml \
+ -o remove-old-ca.yml \
+ -o ... additional opsfiles \
+ --vars-store ./creds.yml \
+ -v ... additional vars
+```
+
+* `nats.tls.ca` property is updated to remove the old CA from the concatenated CAs.
+* The director and health monitor continue to only use new client certificates (for mTLS) that were signed by the new NATS CA. Also, in this step the director and health monitor will start to **ONLY** trust NATS server certificates that were signed by the new CA.
+* The NATS server is updated to use a new certificate (used to serve TLS connections) signed by the new NATS CA. Also, in this step the NATS server will start to **ONLY** trust client certificates (for mTLS) that were signed by the new CA.
+* All components can communicate using the new CA.
+
+`remove-old-ca.yml`
+
+```yaml
+---
+- type: replace
+  path: /instance_groups/name=bosh/properties/nats/tls/ca?
+  value: ((nats_server_tls_2.ca))
+
+- type: replace
+  path: /instance_groups/name=bosh/properties/nats/tls/server?
+  value:
+    certificate: ((nats_server_tls_2.certificate))
+    private_key: ((nats_server_tls_2.private_key))
+
+- type: replace
+  path: /instance_groups/name=bosh/properties/nats/tls/client_ca?
+  value:
+    certificate: ((nats_ca_2.certificate))
+    private_key: ((nats_ca_2.private_key))
+
+- type: replace
```

Comment:
Am I missing something or aren't the certificates that we replace below already replaced in `add-new-ca.yml`?
- `nats/tls/client_ca`
- `nats/tls/director`
- `nats/tls/health_monitor`

Reply:
Yes, they were regenerated in the `add-new-ca.yml` step.  It is just more orderly to still have them replaced in this step, in my opinion.